src/components/LandingPage/Promo/GitStarPromoSection/useGitStars.ts

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from 'react'
+import { getStarCount } from './getStarCount'
+import { getStarFromLocalStorage } from './getStarFromLocalStorage'
+
+export const useGitStars = () => {
+	const [starCount, setStarCount] = useState(0)
+
+	useEffect(() => {
+		try {
+			const localStorageStarCount = getStarFromLocalStorage()
+			if (localStorageStarCount !== null) {
+				setStarCount(localStorageStarCount.starCount)
+			}
+			if (localStorageStarCount === null) {
+				getStarCount().then((count) => {
+					setStarCount(count)
+				})
+			}
+		} catch (error) {
+			setStarCount(0)
+		}
+	}, [])
+
+	const hasStarsToDisplay = starCount !== 0 && !isNaN(starCount)
```

Comment:
Better use isFinite, cause it will also check for Infinity.

Reply:
added